cmd/osm-controller/main.go

```diff
@@ -80,80 +95,124 @@ func main() {
 	flag.StringVar(&opt.podCidr, "pod-cidr", "172.25.0.0/16", "The network ranges from which POD networks are allocated")
 	flag.StringVar(&opt.nodePortRange, "node-port-range", "30000-32767", "A port range to reserve for services with NodePort visibility")
 
+	flag.StringVar(&opt.healthProbeAddress, "health-probe-address", "127.0.0.1:8085", "The address on which the liveness check on /healthz and readiness check on /readyz will be available")
+	flag.StringVar(&opt.metricsAddress, "metrics-address", "127.0.0.1:8080", "The address on which Prometheus metrics will be available under /metrics")
+
 	flag.Parse()
 
 	if len(opt.namespace) == 0 {
 		klog.Fatal("-namespace is required")
 	}
 
+	if len(opt.externalClusterKubeconfig) == 0 {
+		klog.Fatal("-external-cluster-kubeconfig is required")
+	}
+
 	if !(opt.containerRuntime == "docker" || opt.containerRuntime == "containerd") {
 		klog.Fatalf("%s not supported; containerd, docker are the supported container runtimes", opt.containerRuntime)
 	}
 
-	opt.kubeconfig = flag.Lookup("kubeconfig").Value.(flag.Getter).Get().(string)
-
-	// out-of-cluster config was not provided using the flag, try to use the in-cluster config.
-	if opt.kubeconfig == "" {
-		opt.kubeconfig = getKubeConfigPath()
-	}
+	opt.externalClusterKubeconfig = flag.Lookup("external-cluster-kubeconfig").Value.(flag.Getter).Get().(string)
 
 	parsedClusterDNSIPs, err := parseClusterDNSIPs(opt.clusterDNSIPs)
 	if err != nil {
 		klog.Fatalf("invalid cluster dns specified: %v", err)
 	}
 
-	mgr, err := manager.New(config.GetConfigOrDie(), manager.Options{})
+	mgr, err := createManager(opt, opt.externalClusterKubeconfig)
```

Comment:
So as it seems, you are using the worker nodes kubeconfig to create the manager, which means the manager client will be the client for the worker nodes cluster not the seed. Below you are passing the manager to the OSP controller and the osp controller is calling manager.GetClient, which is the client for the worker nodes and that's wrong.

Reply:
The code snippet above shows how we can tackle having two handle two cluster with the same manager.